demux/demux_playlist.c

```diff
@@ -416,6 +416,11 @@ static bool test_path(struct pl_parser *p, char *path, int autocreate)
     if (autocreate & AUTO_ANY)
         return true;
 
+    // When opening a file from cwd, 'path' starts with "./" while stream->path
+    // does not. Strip it to make the comparison work.
+    if (strncmp(path, "./", 2) == 0 && !mp_path_is_absolute(bstr0(p->real_stream->path)))
```

Comment:
I see we convert to bstr below anyway. What do you think about this?

```suggestion
    bstr bpath = bstr0(path);
    if (!mp_path_is_absolute(bstr0(p->real_stream->path)))
        bstr_eatstart0(path, "./");
```

Reply:
I can but this doesn't fix the problem that if you do `mpv ./file` it's still broken anyway.

Reply:
Done and also resolves the `mpv ./file` issue. Not very happy with it but don't have many other ideas besides the other PR I opened